utils/i18nTransformer.js

```diff
@@ -0,0 +1,65 @@
+export default function transformer(file, api) {
+	const j = api.jscodeshift;
+
+  	const basePath = j(file.source);
+
+	const modName = basePath
+  		.find(j.NewExpression, { callee: { name: 'Module' } })
+    	.get().node.arguments[0].value;
+
+	const newI18nKeys = {};
+
+	const modOptionsPath = basePath
+		.find(j.AssignmentExpression, {
+			left: {
+				type: 'MemberExpression',
+				object: { name: 'module' },
+				property: { name: 'options' },
+			},
+		})
+  		.at(0);
+
+	const _ = require("lodash");
+
+	const upCase = function (str) {
+		return str.slice(0,1).toUpperCase() + str.slice(1);
+	}
+
+	modOptionsPath
+		.find(j.Property, { value: { properties: [] } })
+  		.filter(path => path.parentPath.parentPath.parentPath.node === modOptionsPath.get().node)
+        .forEach(optionPath => {
+            const optName = optionPath.node.key.name;
+
+            var titleKey = `${modName}${upCase(optName)}Title`; // use _.camelCase when outside of astexplorer
+
+            newI18nKeys[titleKey] = _.startCase(optName); // use _.startCase
+            optionPath.node.value.properties.unshift(j.property('init', j.identifier('title'), j.literal(titleKey)));
+
+            j(optionPath)
+                .find(j.Property, { kind: 'init', key: { name: 'description' } })
+      			.filter(path => path.parentPath.parentPath.parentPath.node === optionPath.node)
+                .replaceWith(descPath => {
+                    const descKey = `${modName}${upCase(optName)}Desc`; // use _.camelCase
+                    newI18nKeys[descKey] = descPath.node.value.type === 'TemplateLiteral' ?
+                        descPath.node.value.quasis[0].value.raw.trim() :
+                        descPath.node.value.value;
+                    return j.property('init', j.identifier('description'), j.literal(descKey));
+                });
+    	});
+
+			var fs = require('fs');
+			var path = require("path");
+			const FileLocation = path.posix.resolve('./locales/locales/en.json');
+			var enJSON = fs.readFileSync( FileLocation, {'encoding': 'utf8'});
+			var enJSONObj = JSON.parse(enJSON);
+
+			for (var key in newI18nKeys) {
+				enJSONObj[key] = newI18nKeys[key];
+			}
+
+			var newENJSON = JSON.stringify(enJSONObj, null, '\t');
```

Comment:
You could inline a few of these variables that are only used once, but I don't particularly care here